### ISSUE TITLE
twisted/test/test_banana.py uses StringIO incorrectly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2356]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2356|
|Type|defect|
|Created|2007-01-05 15:22:19Z|

The module tries to import cStringIO and falls back to StringIO. It then uses the reset method, which is only defined by the cStringIO version.

<details><summary>Searchable metadata</summary>

```
trac-id__2356 2356
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1168010539000000 1168010539000000
changetime__1176040443000000 1176040443000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

I eliminated the usage of cStringIO in banana-test-fix-2356

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [19967]) Merge banana-test-fix-2356

Author: exarkun
Reviewer: therve
Fixes [#2356](https://github.com/twisted/twisted/issues/2356)

Remove conditional cStringIO usage and cStringIO-specific feature usage from the
banana test suite.  These tests will now actually pass in the absence of cStringIO.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
